### PR TITLE
Admin Activity, Frontend part

### PR DESCRIPTION
**Pull request by GordiNoki**
I'm sorry it took me so long to do that.

### CODE REVIEW

File: apps/frontend/src/app/pages/profile/profile.component.html [hunk 1]
```diff
@@ -10,6 +10,15 @@
             @if (!isLocal) {
               <m-report-button [reportType]="ReportType.USER_PROFILE_REPORT" [reportData]="user.id.toString()" />
             }
+
+            @if (canSeeAdminActivity()) {
+              <div>
```
Comment (tsa96) [inline apps/frontend/src/app/pages/profile/profile.component.html:15]: Pointless div?

File: apps/frontend/src/app/pages/admin/admin-activity/admin-activity.component.ts [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+  activities: { activity: AdminActivity; entry: AdminActivityEntryData }[] = [];
+
+  loading: boolean;
+  readonly pageChange = new Subject<PaginatorState>();
+  readonly filterChange = new Subject<void>();
+
+  readonly rows = 10;
+  totalRecords = 0;
+  first = 0;
+  filter = undefined;
```
Comment (tsa96) [inline apps/frontend/src/app/pages/admin/admin-activity/admin-activity.component.ts:68]: Use `protected` for something the component can see, `private` if not, explicit `public` if it should be visible to parent components. I know I'm nitpicking, but I really like making it clear what's exposes to the component and what isn't.

**Review by tsa96 — CHANGES_REQUESTED** [review]
Nice work! Sorry I'm nitpicking your Angular stuff so much, but since I've been cleaning stuff up so much lately I want to maintain some consistency.

File: apps/frontend/src/app/pages/profile/profile.component.html (hunk 1 above)
Comment (GordiNoki) [reply apps/frontend/src/app/pages/profile/profile.component.html:15]: Yes. Didn't notice that when I was copying Edit Profile button. Should I remove <div> there too?

File: libs/db/src/scripts/seed.ts [hunk 3]
```diff
@@ -231,9 +233,12 @@ prismaWrapper(async (prisma: PrismaClient) => {
...
               axios
-                .get('https://picsum.photos/1920/1080', {
-                  responseType: 'arraybuffer'
-                })
+                .get(
+                  faker.image.urlLoremFlickr({ width: 1920, height: 1080 }),
+                  {
+                    responseType: 'arraybuffer'
+                  }
+                )
```
[the remark this answers is not in the document]
Comment (GordiNoki) [reply libs/db/src/scripts/seed.ts:241]: I forgot to remove that before pushing. picsum throws 403 sometimes in my place, so I change it to lorem flickr

File: apps/frontend/src/app/pages/profile/profile.component.html (hunk 1 above)
Comment (tsa96) [reply apps/frontend/src/app/pages/profile/profile.component.html:15]: Yeah go for it

File: apps/frontend/src/app/pages/admin/admin-activity/admin-activity-entry/admin-activity-entry.component.ts [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+  diffEntries: {
+    key: string;
+    old?: string;
+    new?: string;
+  }[] = [];
+
+  ngOnInit() {
+    if (this.activityData.diff)
+      this.diffEntries = Object.entries(this.activityData.diff).map(
+        ([key, value]) => ({ key, old: value[0], new: value[1] })
```
[the remark this answers is not in the document]
Comment (GordiNoki) [reply apps/frontend/src/app/pages/admin/admin-activity/admin-activity-entry/admin-activity-entry.component.ts:47]: Nope.
image

[the remark this answers is not in the document]
Comment (GordiNoki) [reply apps/frontend/src/app/pages/admin/admin-activity/admin-activity-entry/admin-activity-entry.component.ts:47]: But....
image

File: apps/frontend/src/app/pages/admin/admin-activity/admin-activity-entry/admin-activity-entry.component.ts [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+      );
+  }
+
+  static getActivityData(activity: AdminActivity): AdminActivityEntryData {
```
[the remark this answers is not in the document]
Comment (GordiNoki) [reply apps/frontend/src/app/pages/admin/admin-activity/admin-activity-entry/admin-activity-entry.component.ts:51]: Actually, casting old and new data entry to any in method arguments also works
```ts
static getActivityData(
    activity: AdminActivity & { oldData: any; newData: any }
  ): AdminActivityEntryData {
  ```

File: apps/frontend/src/app/pages/admin/admin-activity/admin-activity-entry/admin-activity-entry.component.ts [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+  standalone: true,
+  imports: [NgStyle, RouterLink]
+})
+export class AdminActivityEntryComponent implements OnInit {
+  @Input({ required: true }) activityData: AdminActivityEntryData;
+
+  diffEntries: {
+    key: string;
+    old?: string;
+    new?: string;
```
[the remark this answers is not in the document]
Comment (GordiNoki) [reply apps/frontend/src/app/pages/admin/admin-activity/admin-activity-entry/admin-activity-entry.component.ts:41]: Agreed, it was an issue with thing above. Fixed

File: apps/frontend/src/app/pages/profile/profile.component.ts [hunk 7]
```diff
@@ -166,4 +168,16 @@ export class ProfileComponent implements OnInit, OnDestroy {
...
+    this.router.navigate([`/admin/admin-activity/${this.user.id}/`]);
+  }
+
+  canSeeAdminActivity(): boolean {
+    return (
+      this.localUserService.hasRole(Role.MODERATOR | Role.ADMIN) &&
+      (this.isLocal ||
+        Bitflags.has(CombinedRoles.MOD_OR_ADMIN, this.user.roles))
+    );
+  }
```
[the remark this answers is not in the document]
Comment (GordiNoki) [reply apps/frontend/src/app/pages/profile/profile.component.ts:182]: Again, I copied the structure of Edit Profile button, which also uses canEdit function:
```ts
  canEdit(): boolean {
    return (
      this.isLocal || this.localUserService.hasRole(Role.MODERATOR | Role.ADMIN)
    );
  }
  ```
  What should I do in that case? Change both?

**Review by tsa96 — APPROVED** [review]
Okay, awesome work! Really happy to get this system in place before 0.10.0.

**Pull request merged**